crates/ruff_linter/src/rules/ruff/rules/float_literal_decimal.rs

```diff
@@ -0,0 +1,76 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast};
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `Decimal` calls passing a float literal.
+///
+/// ## Why is this bad?
+/// Float literals are non-deterministic and can lead to unexpected results. The `Decimal` class is designed to handle
```

Comment:
Non-deterministic isn't the right word here. This means that the computer may use a different value of 0.3. That's not the case. The value for the 0.3 literal is deterministic. 

```suggestion
/// Float literals have limited precision that can lead to unexpected results. The `Decimal` class is designed to handle
```